MonkeyWrench.Web.WebService/WebServices.asmx.cs

```diff
@@ -2633,234 +2633,187 @@ public GetBuildInfoResponse GetBuildInfoMultiple (WebServiceLogin login, string
 
 				lanes = db.GetAllLanes ();
 
-				switch (response.Host.QueueManagement) {
-				case DBQueueManagement.OneRevisionWorkAtATime:
```

Comment:
Instead of replacing the existing code, please add a new enum value to DBQueueManagement and add the new logic there.

This allows you to choose the algorithm for each bot, instead of having to roll out the new algorithm for all bots at the same time (and it's easier to go back to the old algorithm if you find something wrong with your code or logic).